src/main/webapp/js/index/index.js

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* ==========================================================================
+   ROULETTE
+   ========================================================================== */
+function query(queryStr) {
+    const errorEl = document.getElementById("error");
+    fetch(`/query?${queryStr}`, { method: "POST" })
+        .then((response) => response.json())
+        .then((response) => {
+            console.log(response);
+            if (response.status === "OK")
+                redirectToUrl('results.html');
+            else if (response.status === "INVALID_REQUEST") throw "Invalid request";
+            else if (response.status === "ZERO_RESULTS") throw "No results";
+            else if (response.status === "NO_REROLLS") throw "No re-rolls left";
+            else throw "Unforeseen error";
+        })
+        .catch((error) => {
+            errorEl.classList.remove("success-banner");
+            errorEl.classList.remove("hidden");
+            errorEl.classList.add("error-banner");
+            errorEl.innerText = error;
+        });
+}
+
+/* ==========================================================================
+   USER LOCATION AND ADDRESS
+   ========================================================================== */
+function getLocation() {
+    if (navigator.geolocation)
+        navigator.geolocation.getCurrentPosition(geoLocEnabled, geoLocFallback);
+    else
+        geoLocFallback();
+}
+
+// Geolocation is supported and enabled
+function geoLocEnabled(position) {
+    let locationEl = document.getElementById("location-container");
+    let pos = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude,
+    };
+    localStorage.setItem("lat", pos.lat);
+    localStorage.setItem("lng", pos.lng);
+    convertLocation(pos).then((address) => { locationEl.innerText = address; });
+}
+
+// Use inaccurate IP-based geolocation instead
+function geoLocFallback() {
+    let locationEl = document.getElementById("location-container");
+    fetch('https://www.googleapis.com/geolocation/v1/geolocate?key=' + 'AIzaSyBL_9GfCUu7DGDvHdtlM8CaAywE2bVFVJc', {
```

Comment:
this key should have been removed in another PR

Reply:
sure, will sync with secrets manager in another PR

Reply:
or while rebasing